engine/flink/executor/src/main/scala/pl/touk/nussknacker/engine/process/registrar/FlinkProcessRegistrar.scala

```diff
@@ -132,15 +146,22 @@ class FlinkProcessRegistrar(
       val exceptionHandlerPreparer = (runtimeContext: RuntimeContext) =>
         compilerDataForProcessPart(None)(runtimeContext.getUserCodeClassLoader).prepareExceptionHandler(runtimeContext)
       val jobData = compilerData.jobData
+      val engineRuntimeContext =
+        if (isTest) { eng =>
```

Comment:
It would be nice to move this factory logic to the one place where we control how things are created. We already have the `FlinkProcessCompilerDataFactory` which is our IoC container for the flink runtime.

Reply:
It isn't in one place, I see at least four duplicated places. Take a look at comments below